packages/terra-graphs/src/components/Bar/BarGraph.jsx

```diff
@@ -0,0 +1,47 @@
+/* eslint import/no-unresolved: off */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '../Graph.module.scss';
+import './BarGraph.module.scss';
+import utils from '@cerner/carbon-graphs/lib/js/helpers/utils';
+
+const propTypes = {
+  /**
+   * ID for graph created.
+   */
+  graphID: PropTypes.string.isRequired,
+  /**
+   * Graph configuration object
+   */
+  graphConfig: PropTypes.object.isRequired,
+  /**
+   * data to be displayed in graph
+   */
+  dataset: PropTypes.array,
+};
+
+const BarGraph = ({
+  graphConfig, dataset, graphID,
+}) => {
+  React.useEffect(() => {
+    const graph = Carbon.api.graph(graphConfig);
+    if (!(utils.isUndefined(dataset))) {
+      dataset.forEach((data) => {
+        graph.loadContent(Carbon.api.bar(data));
+      });
+    }
+  }, [graphConfig, dataset]);
```

Comment:
It may be worth adding to the docs or to the propType comments that consumers should pass memoized or otherwise consistent object references for the graphConfig/dataset props if the values inside are not changing. Defining those props inline will cause this effect to execute with each update, which will get expensive (I assume?)

Reply:
The array of dependencies on line 34 is used to determine whether the hook should execute for any given update. My point is that if the graph config doesn't change, and the dataset doesn't change, we shouldn't have to reinitialize/reload the graph when the BarGraph component updates. 

React does a reference equality check, not a deep equality check, for each element in the dependency array to determine whether the effect should be executed. So doing something like this:

```jsx
const MyBarGraph = () => (
  <BarGraph
    dataset={[{ key: 'value1' }, { key: 'value2' }]}
  />
);
```

will cause the effect in the BarGraph to execute every time it updates, because a new Array is being instantiated for the dataset every time MyBarGraph updates (even though the raw dataset content has not changed). However, doing something like this:

```jsx
const MyBarGraph = () => {
  const [dataset, setDataSet] = React.useState([{ key: 'value1' }, { key: 'value2' }]);
  
  return (
    <BarGraph
      dataset={dataset}
    />
  );
}
```

will result in a consistent Array reference being provided as the dataset.

Because these props are provided to the BarGraph, it is the responsibility of the BarGraph consumer to ensure these references remain consistent whenever possible. My recommendation is to add some doc (something like "It is recommended to provide memoized or otherwise constant values for this prop to maximize rendering performance") to either the prop descriptions or the doc pages themselves to call this out explicitly).

For simple effects that don't have much of a performance concern, it's usually easier to just let them execute again rather than worry about storing references or memoizing things. For the graphs, though, I assume the rendering cost will be quite high. Correct me if I'm wrong, I don't have much experience with carbon.

More information can be found here regarding effect optimization: https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
Fixed [here](https://github.com/cerner/terra-graphs/pull/106/commits/8f1b7d01eb1f006f648f922b7b0ca25095b52fb2#diff-950a11606d4a7f93ad0d572d6951e24cd4029898d62a96dde1ee82c45bbfa4d1R12). Fixed every example in similar way.